server.js

```diff
@@ -23,16 +39,112 @@ mongoose.Promise = Promise
+})
+
+//SHOWS ALL THE NOMINATIONS
+app.get('/nominations', async (req, res) => {
+  const { category, nominee, film } = req.query
+
+  // POSSIBLE QUERIES: CATEGORY, NOMINEE & FILM
+  const showNominations = await Nomination.find({
+    category: new RegExp(category, 'i'),
+    nominee: new RegExp(nominee, 'i'),
+    film: new RegExp(film, 'i'),
+  })
+
+  if (showNominations) {
+    res.json(showNominations)
+  } else {
+    res.status(404).json({ error: 'Error in search' })
+  }
+})
+
+// SHOW NOMINATION BY ID
+app.get('/nominations/id/:id', async (req, res) => {
+  const nominations = await Nomination.findById(req.params.id)
+  res.json(nominations)
+})
+
+// SHOW ALL NOMINATIONS IN A YEAR
+app.get('/nominations/:year_award', async (req, res) => {
+  const { year_award } = req.params
+  let yearNominations = await Nomination.find({ year_award: year_award })
+
+  // FILTER THE NOMINATIONS IN A YEAR TO ONLY SHOW THE ONES WHO WON
+  const showWin = req.query.win
+
+  if (showWin && yearNominations) {
+    yearNominations = yearNominations.filter((item) => item.win === true)
+  }
+  res.json(yearNominations)
+
+  if (yearNominations) {
+    res.json(yearNominations)
+  } else {
+    res.status(404).json({ error: 'Not found' })
+  }
+})
+
+// SHOW ALL THE NOMINATION IN A SPECIFIC CATEGORY - E.G.: DRAMA, COMEDY, BEST DIRECTOR, SUPPORTING ROLE ETC.
+app.get('/nominations/category/:category', async (req, res) => {
+  const nominations = await Nomination.find()
+  const nominationCategory = req.params.category
+  const showCategoryWin = req.query.win
+
+  let categories = nominations.filter((item) => item.category.includes(nominationCategory))
+
+  // FILTER A SPECIFIC CATEGORY BY WINS
+  if (showCategoryWin && categories) {
+    categories = categories.filter((item) => item.win === true)
+  }
+  res.json(categories)
+
+  if (categories) {
+    res.json(categories)
+  } else {
+    res.status(404).json({ error: 'Not found' })
+  }
+})
+
+// SHOW A SPECIFIC CATEGORY AND A SPECIFIC YEAR
+app.get('/nominations/category/:category/year/:year', async (req, res) => {
+  const nominations = await Nomination.find()
+  const nominationCategory = req.params.category
+  const year = req.params.year
+  const showWins = req.query.win
+
+  let categories = nominations.filter((item) => item.category.includes(nominationCategory))
+
+  // POSSIBLE TO FILTER TO SHOW ONLY WINS
+  if (showWins && categories && year) {
+    let wonCategories = categories.filter((item) => item.win === true)
+    wonCategories = wonCategories.filter((item) => item.year_award === +year)
+    res.json(wonCategories)
+  } else if (categories && year) {
+    const yearCategories = categories.filter((item) => item.year_award === +year)
+    res.json(yearCategories)
+  } else {
+    res.status(404).json({ error: 'Not found' })
+  }
 })
```

Comment:
Here again. Filtering by using mongoose is much easier. You just put all criteria to the find() method.
You can read more here: https://mongoosejs.com/docs/api.html#model_Model.find
And if you still will have a lot of questions just ping me on Slack and we will try to make it clear for you.